### PR TITLE
Apply a guard only on ios

### PR DESCRIPTION
**Pull request by savelichalex**

### CODE REVIEW

File: packages/navigation/src/UILargeTitleHeader/useOnScrollHandler/onScroll.native.tsx [hunk 1]
```diff
@@ -40,15 +42,14 @@ export default function (
...
-         * since the guard is needed only for iOS
+         * The fix is needed only for iOS
          *
          * On iOS `onScroll` event could fire on mount sometimes,
          * that's likely a bug in RN or iOS itself.
          * To prevent changes when there wasn't onBeginDrag event
          * (so it's likely not an actual scroll) using a guard
          */
-        if (ctx != null && !ctx.scrollTouchGuard) {
+        if (isIOS && !ctx != null && !ctx.scrollTouchGuard) {
```
Comment (sertony) [inline packages/navigation/src/UILargeTitleHeader/useOnScrollHandler/onScroll.native.tsx]: Maybe you wanted to do this? I'm not sure why have you added the leading `!` to `ctx != null` condition...
```suggestion
        if (isIOS && ctx != null && !ctx.scrollTouchGuard) {
```

**Review by sertony — CHANGES_REQUESTED** [review]
Consider changing the base into `surf@6.3.0` if it's a bug to fix for beta

**Comment by sertony** [comment]
Btw, what bug does it fix?

**Comment by savelichalex** [comment]
> Btw, what bug does it fix?

So this fix is not needed on Android, but it brakes normal behaviour, as we call `onScroll` without passing a context from a gesture handler (and we can't get this context, moreover we shouldn't)

**Review by sertony — APPROVED** [review]

**Review by SAnatoliiS — APPROVED** [review]

**Review by sertony — APPROVED** [review]

**Pull request merged**